src/libs/styles.c

```diff
@@ -743,8 +747,7 @@ static void _update(dt_lib_module_t *self)
   dt_lib_cancel_postponed_update(self);
   dt_lib_styles_t *d = (dt_lib_styles_t *)self->data;
 
-  const GList *imgs = dt_view_get_images_to_act_on(TRUE, FALSE, FALSE);
-  const gboolean has_act_on = imgs != NULL;
+  const gboolean has_act_on = (dt_act_on_get_images_nb(TRUE, FALSE) > 0);
```

Comment:
... and this here.

And you have a nice refactoring of the act-on in a separate unit I would go even forward and introduce an enum

ACT_ON_ANY
ACT_ON_ONE
ACT_ONE_MULT

And have this returned by a new function (`dt_act_on_get_mode()` or systematically return this value in `dt_act_on_get_images()` by using an access to this enum as parameter).

Reply:
well, in fact, that's not the same : the second one only use "nb of images > 0" so the enum and refactoring would only concern image.c... not sure it's worth the pain... and in fact I thing the current implementation is more clear to read.
That said, I can do the refactoring if you insist, that's not a problem ;)

Reply:
In tagging you also have `dt_act_on_get_images_nb(FALSE, FALSE) > 0` and I'm sure at some point some other checks will be added in many places, so I still think a nice API to get those state would be nice.

Reply:
Fair enough... I'll do that :)